cloud/scope/machine_pool.go

```diff
@@ -150,6 +166,39 @@ func (m *MachinePoolScope) GetWorkerMachineSubnet() *string {
 	return nil
 }
 
+// ListMachinePoolInstances returns the WorkerRole core.Subnet id for the cluster
+func (m *MachinePoolScope) ListMachinePoolInstances(ctx context.Context) ([]core.InstanceSummary, error) {
+	poolOid := m.GetPoolOCID()
+
+	req := core.ListInstancePoolInstancesRequest{
+		CompartmentId:  common.String(m.OCICluster.Spec.CompartmentId),
+		InstancePoolId: common.String(poolOid),
+	}
+	resp, err := m.ComputeManagementClient.ListInstancePoolInstances(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.Items, nil
+}
+
+// SetListandSetMachinePoolInstances retrieves a machine pools instances and sets them in the ProviderIDList
+func (m *MachinePoolScope) SetListandSetMachinePoolInstances(ctx context.Context) (int32, error) {
+	poolInstanceSummaries, err := m.ListMachinePoolInstances(ctx)
+	if err != nil {
+		return 0, errors.New("Unable to list machine pool's instances")
+	}
+	providerIDList := make([]string, len(poolInstanceSummaries))
+
+	for i, instance := range poolInstanceSummaries {
+		providerIDList[i] = fmt.Sprintf("oci://%s/%s/%s", *instance.Id, *instance.AvailabilityDomain, *instance.FaultDomain)
```

Comment:
provider id does not have ad/fd etc right? we dont do that in machine, it is just oci://<instance-ocid> . Any reason why you added this?